modules/light-clients/06-solomachine/solomachine_test.go

```diff
@@ -46,7 +54,7 @@ func TestSoloMachineTestSuite(t *testing.T) {
 	suite.Run(t, new(SoloMachineTestSuite))
 }
 
-func (suite *SoloMachineTestSuite) TestSolomachineSetup() {
+func (suite *SoloMachineTestSuite) SetupSolomachine() string {
```

Comment:
I think we should maybe return a `SolomachineConfig` struct from this function which encapsulates the client/connection/channel/port IDs.

Reply:
looking into this now, is looks like the endpoints in all the other tests are constructed using a `suite.chainA` and `suite.chainB`, for this will we need to fully create instances of `ibctesting.Endpoint` that are referencing `solomachine` and `chainA`?

Reply:
@colin-axner do you think we should do this in this PR or are you happy with a follow up issue? This PR is already pretty big!

Reply:
I think any change to use endpoint can be made in a followup pr. I would not recommend modifying the testing package to handle solo machine endpoints, I was primarily thinking that you could manually fill in the endpoint with the correct values without supporting the ability to do `suite.coordinator.Setup(path)`. I see the difficulty though since when you call `NewEndpoint` you provide quite a bit of information. What do you think about having a `NewSolomachineEndpoint`, also very open to just creating a new solomachine endpoint struct which contains less information (if that makes sense). I figure we can always consolidate the two types in the future if there is enough overlap

Reply:
I think a new type may work well here, I will create a follow up issue for this and look into after this is merged into your branch! 👍